File: src/quiz_api.py
from requests import get

class QuizAPI:
	def __init__(self, api_key: str) -> None:
		self.api = "https://quizapi.io/api"
		self.api_key = api_key
		self.headers = {
			"user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36",
			"x-api-key": self.api_key
		}

	def get_random_quiz(self, limit: int = 1) -> dict:
		return get(
			f"{self.api}/v1/questions?limit={limit}",
			headers=self.headers).json()

	def get_quiz_by_category(
			self,
			category: str,
			limit: int = 1) -> dict:
		return get(
			f"{self.api}/v1/questions?category={category}&limit={limit}",
			headers=self.headers).json()

	def get_quiz_by_difficulty(
			self,
			difficulty: str,
			limit: int = 1) -> dict:
		return get(
			f"{self.api}/v1/questions?difficulty={difficulty}&limit={limit}",
			headers=self.headers).json()

	def get_quiz_by_tags(
			self,
			tags: str,
			limit: int = 1) -> dict:
		return get(
			f"{self.api}/v1/questions?tags={tags}&limit={limit}",
			headers=self.headers).json()

File: src/quiz_api..js
class QuizApi {
	constructor(apiKey) {
		this.api = "https://quizapi.io/api/v1"
		this.apiKey = apiKey
		this.headers = {
			"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36",
			"x-api-key": this.apiKey
		}
	}

	async getRandomQuiz(limit = 1) {
		const response = await fetch(
			`${this.api}/questions?limit=${limit}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getQuizByCategory(category, limit = 1) {
		const response = await fetch(
			`${this.api}/questions?category=${category}&limit=${limit}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getQuizByDifficulty(difficulty, limit = 1) {
		const response = await fetch(
			`${this.api}/questions?difficulty=${difficulty}&limit=${limit}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getQuizByTags(tags, limit = 1) {
		const response = await fetch(
			`${this.api}/questions?tags=${tags}&limit=${limit}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {QuizApi}
